src/setup.jl

```diff
@@ -1,16 +1,17 @@
 module Setup
 
 using Pkg: GitTools
-import Artifacts: @artifact_str
+import Artifacts: find_artifacts_toml, @artifact_str
 import GAP_jll
 import GAP_lib_jll
 import GAP_pkg_juliainterface_jll
 import Scratch: @get_scratch!
 import Pidfile
+import TOML
 
 # to separate the scratchspaces of different GAP.jl copies and Julia versions
 # put the Julia version and the hash of the path to this file into the key
-const scratch_key = "gap_$(string(hash(@__FILE__)))_$(VERSION.major).$(VERSION.minor)"
+const scratch_key = "gap_$(hash(@__FILE__))-$(VERSION.major).$(VERSION.minor)"
```

Comment:
IIuc this change is here to not use the same scratchspace as before as that could lead to conflicts, right? Wouldn't it make sense to either include some kind of increasing number in either the key or in the hash input, so we can bump that again for the next incompatible change in the future?

Reply:
I actually had an intermediate version where I computed a more complicated hash, which involved the paths of the various artifacts (!) as those can also change; but also checksums of e.g. `setup.jl` and other files, but then realized this doesn't solve all the problems either... 

Here are some further alternatives:
- as above, include more data into the hash (e.g. GAP.jl version, location of the various JLLs and artifacts we link to, ...)
- stop using a scratch space, instead just create a fresh "mutable root dir" each time the package is loaded in a fresh temp dir (of course that means a *lot* of temp dirs would accumulate... we could add an `atexit` handler to delete them at the end?)
   - would be interesting how much time we spend creating the mutable root dir
- we could adjust the setup code so that whenever it e.g. creates a dir or symlink it checks if there is already an object of the correct type there -- i.e. if we created a dir `pkg` and there is already a dir `pkg`, do nothing; but if we create a symlink `pkg` pointing to `targetA` and there is instead a directory `pkg`; or a symlink pointing to `anotherTarget`, then delete `pkg`.
  - this seems much more complicated and in every way I can think of inferior to the solution of just rebuilding it all from scratch, *except* that at least the path to this root dir is somewhat stable and predictable
  - strictly speaking we'd also have to scan for obsolete file system objects that should be removed... making this even more complicated...

Reply:
See also https://github.com/oscar-system/GAP.jl/issues/993